### ISSUE TITLE
FEATURE: onyo new creating a duplicate/clone (with new serial)

### ISSUE THREAD

**Issue by j-leo** [psyinfra/onyo#461]
In the past while managing inventory we often used the clone feature to create a duplicate of an asset. Here in Onyo we should make sure that the serial needs to be redefined during this process. By default it could generate a faux serial if no new one was given.

**Comment by j-leo** [comment]
even better: make it possible to create N clones

**Comment by bpoldrack** [comment]
@j-leo :

In essence this sounds like a template. Except that it would be created from an existing asset.
Wonder whether it's worth inflating the interface of `new` with a `--from` or `--clone` switch, or maybe simply have a `onyo template --from <path> --to <template_name> --keys <key>=<value>` where one can specify which fields to override (in your case `serial=faux`). Result is a new template, that one can reference in a subsequent `new` call.
WDYT?

**Comment by j-leo** [comment]
Your option sounds useful in addition :)

I am afraid of ending up with too many templates, and I wanted to have a quick possibility. But yeah, in the end the question is, when is this needed AND a template does not make sense?

Maybe we start with your idea and see how reality works out.

**Comment by TobiasKadelka** [comment]
Strongly connected to #306!

I like your proposed `onyo template --from <path> --to <template_name> --keys <key>=<value>` and think it could/should solve multiple scenarios, and help the efforts described in #306!

Syntax wise I propose:
- not even use `--from/to`, and instead have the first path as source, the second the destination name, like with `cp` or `mv`.
- If just one is provided, it is the name of the new template (e.g. no source is used).
- More than two should error.
- `--keys` makes very much sense!
- The other flag I think would be valuable is `onyo template --edit`, in parallel to `onyo new --edit`

**Comment by bpoldrack** [comment]
So, I would agree - let's go for a `template` command first. If we need the "quick" option, we could later allow `new` to take any asset as a template and generate an "ad-hoc template" that isn't stored (taking that functionality from the `template` command)

**Comment by bpoldrack** [comment]
Verdict from discussion in dev chat:
- Introduce `--clone` for `new` (PR #551)
- Make separate issue for a possible `template` command.

Hence, this will be closed by merging #551.
